codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocol.kt

```diff
@@ -37,12 +36,10 @@ import software.amazon.smithy.rust.codegen.server.smithy.protocols.ServerAwsJson
 import software.amazon.smithy.rust.codegen.server.smithy.protocols.ServerRestJsonSerializerGenerator
 import software.amazon.smithy.rust.codegen.server.smithy.targetCanReachConstrainedShape
 
-private fun allOperations(codegenContext: CodegenContext): List<OperationShape> {
-    val index = TopDownIndex.of(codegenContext.model)
-    return index.getContainedOperations(codegenContext.serviceShape).sortedBy { it.id }
-}
-
 interface ServerProtocol : Protocol {
+    /** The path such that `aws_smithy_http_server::proto::$path` points to the protocol's module. */
+    val protocolModulePath: String
```

Comment:
Hmm the `ServerProtocol` should know where these types lie, we shouldn't invert the responsibility on the caller in `ServerRuntimeType`. `protocolModulePath` is just to make it so that most implementers can rely on the default implementations, but AWS JSON needs to switch based on version.

Reply:
> `protocolModulePath` is just to make it so that most implementers can rely on the default implementations

Right, in Rust this would be akin to adding a `const DEFAULT: &'static str;` to a trait in order to get default function implementations to work. In the case where functions don't follow the default implementation this field becomes useless.

Reply:
Maybe I'm just misunderstood Kotlin here, is the function implementation overridable?

```kotlin
fun requestRejection(runtimeConfig: RuntimeConfig): RuntimeType =
        ServerCargoDependency.smithyHttpServer(runtimeConfig)
            .toType().resolve("proto::$protocolModulePath::rejection::RequestRejection")
```

If not then I actually have no problem with this - we're tightening the `ServerProtocol` contract but there's no redundancy.

Reply:
It is overridable. `ServerAwsJsonProtocol` overrides it.

Reply:
> In the case where functions don't follow the default implementation this field becomes useless?

Not useless, the field can be used as implementations see fit. See e.g. the implementation of `markerStruct` in `ServerRestJsonProtocol`.

Reply:
It is used by the default function impls.

Reply:
I mean in the case we are not using the default function impls* then the `val` can live in the class implementing the interface.